File: src/types/ffmpeg.ts
export type FFmpegSource = 'System' | 'Bundled';

export interface FFmpegStatus {
    installed: boolean;
    source: FFmpegSource;
}

export interface FFmpegConfig {
    ffmpeg_path: string | null;
    ffprobe_path: string | null;
    source: FFmpegSource;
}

export interface DownloadProgress {
    current: number;
    total: number;
    percentage: number;
}

File: src-tauri/src/types/ffmpeg.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FFmpegConfig {
    pub ffmpeg_path: Option<String>,
    pub ffprobe_path: Option<String>,
    pub source: FFmpegSource,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub enum FFmpegSource {
    System,
    Bundled,
}

#[derive(Debug, Clone, Serialize)]
pub struct FFmpegStatus {
    pub installed: bool,
    pub source: FFmpegSource,
}

#[derive(Debug, Clone, Serialize)]
#[allow(dead_code)]
pub struct DownloadProgress {
    pub current: u64,
    pub total: u64,
    pub percentage: u8,
}
